lib/hexpm/accounts/web_auth_request.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Hexpm.Accounts.WebAuthRequest do
+  use Hexpm.Schema
+
+  @moduledoc false
+
+  alias Hexpm.Accounts.Keys
+
+  # See `lib/hexpm/accounts/web_auth.ex` for an explanation of the terminology used.
+
+  @scopes ["read", "write"]
+  @key_permission %{domain: "api", resource: nil}
+  @key_params %{name: nil, permissions: [@key_permission]}
+
+  schema "requests" do
+    field :device_code, :string
+    field :user_code, :string
+    field :scope, :string
+    field :verified, :boolean, default: false
+    belongs_to :user, Hexpm.Accounts.User
+    field :audit, :string
```

Comment:
Is this okay for audit data? For testing we use a string, but I am not so sure if you do so in production?

Reply:
> I don't think authenticating through the web API requires an audit log though, we usually only log actions that change something in the system.

@ericmj, I need the Audit information to generate a key.

> Audit information is stored separately in the AuditLog schema.

I believe for tests that this is stored as a tuple in the form `{%Hexpm.Accounts.User, "test"}`.
Should I submit/work on a PR to fix this in the tests?

Reply:
> I need the Audit information to generate a key.

Is there an issue getting the audit information when generating the key?

Reply:
Ping. Let me know if you need any help with this.

Reply:
Judging from this:

https://github.com/hexpm/hexpm/blob/bc042cacab11cdf27554bfdb06cddc6311c6dbed/lib/hexpm/accounts/audit_log.ex#L68

I should be able to manage with passing nil as audit data?

Reply:
@ericmj,

I don't think this is an issue. The field `audit` stores the `user_agent`, which is a string. I actually was unaware of this and I think I confused every one by calling this field `audit`.

So I have a question: Should I use the browser user agent or the hex user agent? i.e. use the user agent from the submit or the access_key action?

I think we should use the latter. If we decide to use the browser `user_agent`, I will rename the `audit` field to `user_agent`.

Reply:
@ericmj and @wojtekmach,

Ping. I am going to go ahead and use the Hex client `user_agent`.